src/Microsoft.NET.Sdk.Functions.Generator/Program.cs

```diff
@@ -25,7 +26,29 @@ private static void Main(string[] args)
 
                 AssemblyLoadContext.Default.Resolving += (context, assemblyName) =>
                 {
-                    return context.LoadFromAssemblyPath(Path.Combine(assemblyDir, assemblyName.Name + ".dll"));
+                    string assemblyPath = Path.Combine(assemblyDir, assemblyName.Name + ".dll");
+
+                    if (File.Exists(assemblyPath))
+                    {
+                        Assembly assembly = AssemblyLoadContext.Default.LoadFromAssemblyPath(assemblyPath);
+                        return assembly;
+                    }
+
+                    try
+                    {
+                        // If the assembly file is not found, it may be a runtime assembly for a different
+                        // runtime version (i.e. the Function app assembly targets .NET Core 3.1, yet this
+                        // process is running 3.0). In that case, just try to return the currently-loaded assembly,
+                        // even if it's the wrong version; we won't be running it, just reflecting.
+                        Assembly assembly = Assembly.Load(assemblyName.Name);
+                        return assembly;
+                    }
+                    catch (Exception)
+                    {
+                        // We'll already log an error if this happens; this gives a little more details if debug is enabled.
+                        logger.LogError($"Unable to find fallback for assembly '{assemblyName}'.");
+                        return null;
+                    }
```

Comment:
Note this logic all came from EMG: https://github.com/Azure/azure-functions-host/blob/dev/tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/Program.cs#L65-L86